### ISSUE TITLE
Cannot create JSON files

### ISSUE THREAD

**Issue by boulderwebdev** [solid/node-solid-server#1011]
If I try to `POST` or `PUT` a JSON file using `solid.auth.fetch` then an empty file is stored on the server. For example, I could call

```javascript
const data = JSON.stringify({
  "hello": "world"
})
const url = 'https://boulderwebdev.solid.community/public/'
solid.auth.fetch(url, {
  method: "POST",
  headers: {
    "Content-Type": "application/json",
    "Slug": "data",
    "Content-Length": data.length
  },
  body: data
}).then((resp) => {  ... })
```
and the server will create a file and respond with

```
201
allow OPTIONS, HEAD, GET, PATCH, POST, PUT, DELETE
content-length 7
content-type text/plain; charset=utf-8
etag W/"7-rM9AyJuqT6iOan/xHh+AW+7K/T8"
link <.acl>; rel="acl", <.meta>; rel="describedBy", <http://www.w3.org/ns/ldp#Container>; rel="type", <http://www.w3.org/ns/ldp#BasicContainer>; rel="type"
location /public/data.json
vary Accept, Authorization, Origin
```

**Comment by boulderwebdev** [comment]
@kjetilk This is also true for uploading JPEG and PNG files :'(

**Comment by kjetilk** [comment]
@RubenVerborgh Do you think this might be a solid-auth-client issue?

**Comment by RubenVerborgh** [comment]
I'm suspicious of:
- slug being equal to `data`, since it results in an invalid slug (and NSS probably doesn't check)
- `Content-Length` being set in what I presume is the browser (don't do that, the browser does it for you)

**Comment by boulderwebdev** [comment]
@RubenVerborgh I was using the slug `data` as just example data. In my case I am using either a uuid or a file named `config`. I removed the Content-Length header too, but that didn't change anything.

**Comment by RubenVerborgh** [comment]
Okay, thanks. The client still wouldn't be my first suspect, because it doesn't touch any of the relevant headers, nor `body`.

**Comment by boulderwebdev** [comment]
@RubenVerborgh Also, if you set the content type to the invalid value "text/json", the server saves the JSON file correctly...

**Comment by RubenVerborgh** [comment]
Just to be sure

> then an empty file is stored on the server.

So you get the 201, which is empty (that is expected).
But when you perform another GET on the resource, do you receive 200 or 404?

**Comment by bourgeoa** [comment]
I have the same issue trying to add a file.json with the UI green + button. The file is created with zero length. I can create/edit the content with the Solid-ide-file app.
By the way file.js are accepted but the extension is .js_false

**Comment by timbl** [comment]
That is a separate issue with the data browser UI... when you create a new text file  by clicking on the plus, it creates it with zero bytes,.  That's normal. What is non-ideal is that it doesn't open the source pane to allow you to edit it.   (or are you dragging a file onto the green plus?)

**Comment by jaxoncreed** [comment]
Fixed in #1158
